src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustDetailsValidator.java

```diff
@@ -66,17 +71,71 @@ private boolean validateName(String trustName, Errors errors, String loggingCont
+            // Cease date of the trust must be 'null' as there are still Individual and/or Corporate BOs associated with this trust
+
+            if (Objects.nonNull(trustCeaseDate)) {
+                final String errorMessage = ValidationMessages.NULL_ERROR_MESSAGE.replace("%s", qualifiedFieldName);
+                setErrorMsgToLocation(errors, qualifiedFieldName, errorMessage);
+                ApiLogger.infoContext(loggingContext, errorMessage);
+            }
+        } else {
+            final LocalDate creationDate = trustDataDto.getCreationDate();
+
+            // Cease date of the trust cannot be 'null' as there are no longer any Individual or Corporate BOs associated with this trust
+
+            if (UtilsValidators.isNotNull(trustCeaseDate, qualifiedFieldName, errors, loggingContext)) {
+                DateValidators.isDateInPast(trustCeaseDate, qualifiedFieldName, errors, loggingContext);
+                DateValidators.isCeasedDateOnOrAfterCreationDate(trustCeaseDate, creationDate, qualifiedFieldName, errors, loggingContext);
+            }
+        }
+    }
+
+    private boolean isTrustStillRequired(OverseasEntitySubmissionDto overseasEntitySubmissionDto, TrustDataDto trustDataDto) {
+        // Use the trust id on TrustDataDto to whizz through all the Individual and Corporate BOs that are associated
+        // with this trust. As soon as a matching BO is found where cease date is null and a trust NOC is set then this
+        // indicates that the trust is still required
+
+        if (trustDataDto.getTrustId() == null) {
+            return false;   // No trust ID, so trust is not required
+        }
+
+        if (overseasEntitySubmissionDto.getBeneficialOwnersIndividual() != null
+                && overseasEntitySubmissionDto.getBeneficialOwnersIndividual().stream().anyMatch(
+                        boIndividualDto -> boIndividualDto.getTrustIds() != null
+                            && boIndividualDto.getTrustIds().contains(trustDataDto.getTrustId())
+                            && boIndividualDto.getCeasedDate() == null
+                            && boIndividualDto.getTrusteesNatureOfControlTypes() != null
+                            && !boIndividualDto.getTrusteesNatureOfControlTypes().isEmpty())) {
+            return true;
+        }
+
+        return (overseasEntitySubmissionDto.getBeneficialOwnersCorporate() != null
+                && overseasEntitySubmissionDto.getBeneficialOwnersCorporate().stream().anyMatch(
+                        boCorporateDto -> boCorporateDto.getTrustIds() != null
+                            && boCorporateDto.getTrustIds().contains(trustDataDto.getTrustId())
+                            && boCorporateDto.getCeasedDate() == null
+                            && boCorporateDto.getTrusteesNatureOfControlTypes() != null
+                            && !boCorporateDto.getTrusteesNatureOfControlTypes().isEmpty()));
```

Comment:
Shouldn't this just return false? Would it get past the previous block if it returned true? If so, do you need the previous block?

Reply:
Block needed as discussed but I'll make things clearer with the comments.